File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationRouteEventListener.java
package com.mapbox.services.android.navigation.v5.internal.navigation;

import okhttp3.Call;
import okhttp3.EventListener;

public class NavigationRouteEventListener extends EventListener {

  private final ElapsedTime time;

  public NavigationRouteEventListener() {
    this(new ElapsedTime());
  }

  NavigationRouteEventListener(ElapsedTime time) {
    this.time = time;
  }

  @Override
  public void callStart(Call call) {
    super.callStart(call);
    time.start();
  }

  @Override
  public void callEnd(Call call) {
    super.callEnd(call);
    time.end();
  }

  ElapsedTime getTime() {
    return time;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationRouteEventListener.kt
package com.mapbox.services.android.navigation.v5.internal.navigation

import okhttp3.Call
import okhttp3.EventListener

class NavigationRouteEventListener
@JvmOverloads
constructor(
    val time: ElapsedTime = ElapsedTime()
) : EventListener() {

    override fun callStart(call: Call) {
        super.callStart(call)
        time.start()
    }

    override fun callEnd(call: Call) {
        super.callEnd(call)
        time.end()
    }
}
